codemods/videos.js

```diff
@@ -1,22 +1,36 @@
 const visit = require('unist-util-visit');
 const {
   addAttribute,
-  hasClassName,
   findAttribute,
+  isMdxBlockElement,
 } = require('./utils/mdxast');
 
+const getVideoProps = (src) => {
+  const propsObj = {};
+  const domain = src.split('/')[2];
+  if (domain === 'fast.wistia.net') {
+    propsObj.type = 'wistia';
+    propsObj.id = src.match(/iframe\/([a-zA-Z0-9]+)\??/)[1];
+  }
+  if (domain === 'www.youtube.com') {
+    propsObj.type = 'youtube';
+    propsObj.id = src.match(/embed\/([a-zA-Z0-9]+)\??/)[1];
+  } else {
+    throw new Error('Video type not recognized.');
```

Comment:
While I think of it, you might actually consider using the [`vfile` message API](https://github.com/vfile/vfile#vfilemessagereason-position-origin) (`message`, `info`, or `fail` for this error. The nice side effect of that is that you can point to a specific location in the AST. and have that print nicely to the screen. The `file` comes through as the second prop to the plugin function below:

```
const videos = () => (tree, file) => {
  // ...
  visit(..., node => {
    file.fail(new Error(`Video type not recognized for url: ${url}`), node.position)
  })
}
```

Reply:
ah i was going to ask about how to use vfile messaging API and spaced it. yeah i like that. making change now. thanks!